src/qs_apt_fdiff_methods.F

```diff
@@ -0,0 +1,281 @@
+      TYPE(apt_fdiff_points_type), POINTER               :: apt_fdiff_points
+      REAL(kind=dp), DIMENSION(:, :, :)                  :: ap_tensor
+      INTEGER, INTENT(IN)                                :: natoms
+
+      INTEGER                                            :: i, j, n
+
+      DO j = 1, 3 ! axis force
+         DO i = 1, 3 ! axis field
+            DO n = 1, natoms
+               ap_tensor(n, i, j) = (apt_fdiff_points%point_field(i, 1)%forces(n, j) - &
+                                     apt_fdiff_points%point_field(i, 2)%forces(n, j)) &
+                                    /(2*apt_fdiff_points%field_strength)
+            END DO
+         END DO
+      END DO
+
+   END SUBROUTINE fdiff_2pnt
+
+   ! **************************************************************************************************
+   !> \brief ...
+   !> \param force_env ...
+   !> \param apt_fdiff_point ...
+   !> \author Leo Decking
+   ! **************************************************************************************************
+! **************************************************************************************************
+!> \brief ...
+!> \param apt_fdiff_point ...
+!> \param particles ...
+! **************************************************************************************************
+   SUBROUTINE get_forces(apt_fdiff_point, particles)
+      TYPE(apt_fdiff_point_type)                         :: apt_fdiff_point
+      TYPE(particle_list_type), POINTER                  :: particles
+
+      INTEGER                                            :: i
+
+      CPASSERT(ASSOCIATED(particles))
+
+      DO i = 1, particles%n_els
+         apt_fdiff_point%forces(i, 1:3) = particles%els(i)%f(1:3)
+      END DO
+
+   END SUBROUTINE get_forces
+
+   ! **************************************************************************************************
+   !> \brief Calculate Atomic Polarization Tensors by dF/d(E-field) finite differences
+   !> \author Leo Decking, Hossam Elgabarty
+   ! **************************************************************************************************
+! **************************************************************************************************
+!> \brief ...
+!> \param force_env ...
+! **************************************************************************************************
+   SUBROUTINE apt_fdiff(force_env)
+
+      TYPE(force_env_type), POINTER                      :: force_env
+
+      CHARACTER(LEN=*), PARAMETER                        :: routineN = 'apt_fdiff'
+
+      INTEGER                                            :: apt_log, fd_method, handle, i, j, &
+                                                            log_unit, n, natoms, output_fdiff_scf
+      REAL(kind=dp)                                      :: born_sum
+      REAL(kind=dp), ALLOCATABLE, DIMENSION(:, :, :)     :: ap_tensor
+      TYPE(apt_fdiff_points_type), POINTER               :: apt_fdiff_points
```

Comment:
This variable is only used within this module and does not need to be saved anywhere. So the pointer attribute is unnecessary.